source/common/http/http3/conn_pool.h

```diff
@@ -54,10 +59,18 @@ class ActiveClient : public MultiplexedActiveClientBase {
     quiche_capacity_ = new_quiche_capacity;
     uint64_t new_capacity = currentUnusedCapacity();
 
-    if (new_capacity < old_capacity) {
-      parent_.decrClusterStreamCapacity(old_capacity - new_capacity);
-    } else if (old_capacity < new_capacity) {
-      parent_.incrClusterStreamCapacity(new_capacity - old_capacity);
+    if (connect_timer_) {
```

Comment:
Question: Why doesn't this just call `MultiplexedActiveClientBase::onSettings()`? In other words, why is this handled completely differently than http/2?

Reply:
I'm not sure if this answers your question but, I think the crux of the issue is that with HTTP/2 the setting SETTINGS_MAX_CONCURRENT_STREAMS specifies a limit on "open" streams. This means that a client may open a new stream any time it closes a stream. But HTTP/3 stream limits are managed at the QUIC layer. With QUIC, the limit is (effectively) the largest stream ID that can be opened. When a client closes a stream, that has no effect on the limit and the client may not be able to open a new stream. Only when the server sends a new MAX_STREAMS frame will the limit increase. This difference between HTTP/2 and HTTP/3 is pretty fundamental, albeit rather annoying.

Reply:
Got it; that makes sense. Thanks!